tests/vmi_configuration_test.go

```diff
@@ -431,6 +431,84 @@ var _ = Describe("Configurations", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("with guestAgent", func() {
+			var agentVMI *v1.VirtualMachineInstance
+
+			It("should have attached an guest agent channel by default", func() {
+
+				agentVMI = tests.NewRandomVMIWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+				By("Starting a VirtualMachineInstance")
+				agentVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(agentVMI)
+				Expect(err).ToNot(HaveOccurred(), "Should create VMI successfully")
+				tests.WaitForSuccessfulVMIStart(agentVMI)
+
+				getOptions := metav1.GetOptions{}
+				var freshVMI *v1.VirtualMachineInstance
+
+				freshVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, &getOptions)
+				Expect(err).ToNot(HaveOccurred(), "Should get VMI ")
+
+				domXML := getVmDomainXml(virtClient, freshVMI)
+
+				Expect(domXML).To(ContainSubstring("<channel type='unix'>"), "Should contain at least one channel")
+				Expect(domXML).To(ContainSubstring("<target type='virtio' name='org.qemu.guest_agent.0' state='disconnected'/>"), "Should have guest agent channel present")
+				Expect(domXML).To(ContainSubstring("<alias name='channel0'/>"), "Should have guest channel present")
```

Comment:
The indentation is predictable, so you can just test that the channel is, in fact the right one and the lines go one after another. But this should be enough as well.